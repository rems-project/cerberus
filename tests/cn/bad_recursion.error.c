/*@

datatype bad { Bad { cn_map<u8, datatype bad> b } }


@*/

