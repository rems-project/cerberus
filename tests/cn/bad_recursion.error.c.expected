return code: 1
tests/cn/bad_recursion.error.c:3:1: error: Illegal datatype definition.
Constructor argument 'b' is given type 'map<u8, datatype bad>', which indirectly refers to 'datatype bad'.
Indirect recursion via map, set, record, or tuple types is not permitted.
datatype bad { Bad { map<u8, datatype bad> b } }
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
